### PR TITLE
Integrate advanced search into list views

### PR DESCRIPTION
**Pull request by tsauerwein**
This PR adds support for search parameters to the list views. For example for an URL like `/waypoints?wt=summit&we=4500&a=14274,14270&pl=fr`, a query for ElasticSearch will be created and the matching documents are returned (in this case all summits higher than 4500 meters in France and Italy with French locales).

Still to do:

* [x] Add 'sort by' to the ES query
* [ ] Add support for spatial queries (in a new PR)

Closes https://github.com/c2corg/v6_api/issues/29
Closes https://github.com/c2corg/v6_api/issues/197

### CODE REVIEW

**Comment by tsauerwein** [comment]
This PR would be ready to review. I will handle the spatial search in a new PR.

**Comment by gberaudo** [comment]
Will it still work with Elasticsearch 1.6.0 (Ubuntu)?

**Comment by tsauerwein** [comment]
> Will it still work with Elasticsearch 1.6.0 (Ubuntu)?

Likely not. I am planning to use a version >= 2.2 because the spatial support has improved. But you can easily install a more recent version: https://www.elastic.co/downloads/elasticsearch

File: c2corg_api/search/mappings/waypoint_mapping.py [hunk 1]
```diff
@@ -1,11 +1,67 @@
...
     def to_search_document(document, index):
-        return SearchDocument.to_search_document(document, index)
+        search_document = SearchDocument.to_search_document(document, index)
+
+        if document.redirects_to:
+            return search_document
+
+        SearchDocument.copy_fields(
+            search_document, document, SearchWaypoint.FIELDS)
+        search_document['has_phone'] = not (not (
```
Comment (gberaudo) [inline c2corg_api/search/mappings/waypoint_mapping.py]: Is the double not necessary?

Comment (tsauerwein) [inline c2corg_api/search/mappings/waypoint_mapping.py]: Yes. In Python `a or b` gives you `a` if `a` is a *truthy* value. By doing `not(a or b)` I get a boolean value, which I inverse again.

File: c2corg_api/tests/views/test_outing.py [hunk 2]
```diff
@@ -41,6 +43,8 @@ def test_get_collection(self):
         self.assertEqual(author['user_id'], self.global_userids['contributor'])
         self._add_test_data()
 
+    # TODO check how these two requests are used (as part of advanced search?)
+    @unittest.skip
```
Comment (gberaudo) [inline c2corg_api/tests/views/test_outing.py]: Do we still need these two methods?

**Comment by gberaudo** [comment]
Great job @tsauerwein! I added some minor comments, feel free to merge when Travis is happy.
Regarding the version of elasticsearch, +1 to switch to ES 2.2.

**Comment by asaunier** [comment]
> For example for an URL like /waypoints?wt=summit&we=4500&a=14274,14270&pl=fr, a query for ElasticSearch will be created and the matching documents are returned (in this case all summits higher than 4500 meters in France and Italy with French locales).

Nice!
I guess the UI will have to add the parameters to the API request depending on the selected filters.
I guess also that those parameters will be added to the advanced search page URL as in http://www.camptocamp.org/routes/list/areas/14442-14421/malt/%3E3000/act/1

I think the client expects the parameters to be as close as possible to the parameters used in v5. See https://docs.google.com/spreadsheets/d/1LTE1UIjsDTdZauDgR4rKe2UCtXF6LG-ELH53Qwy1nmI/edit#gid=0

**Comment by tsauerwein** [comment]
Ok for merging?

> I think the client expects the parameters to be as close as possible to the parameters used in v5

Alex will check this, if required we can still change the names.

**Comment by asaunier** [comment]
Sure, please merge.

**Comment by gberaudo** [comment]
+1

**Pull request merged**